src/Modal/Modal.js

```diff
@@ -38,163 +49,116 @@ const Backdrop = createComponent({
 });
 
-class Modal extends React.Component {
-  static propTypes = {
-    open: PropTypes.bool,
-    closeOnBackdropClick: PropTypes.bool,
-    closeOnEscape: PropTypes.bool,
-    minWidth: PropTypes.number,
-    maxWidth: PropTypes.number,
-    animationIn: PropTypes.string,
-    animationOut: PropTypes.string,
-    onClose: PropTypes.func,
-  };
+function Modal({ children, title, animationDuration, showClose, onClose, open, ...props }) {
+  const [isOpen, setOpen] = useState(open);
 
-  static defaultProps = {
-    open: false,
-    closeOnBackdropClick: true,
-    closeOnEscape: true,
-    animationIn: 'zoomIn',
-    animationOut: 'zoomOut',
-    onClose: () => {},
-  };
-
-  state = {
-    open: this.props.open || false,
-    opening: false,
-    closing: false,
-  };
-
-  componentDidUpdate(oldProps) {
-    if (!oldProps.open && this.props.open) {
-      this.open();
-    } else if (oldProps.open && !this.props.open) {
-      this.close();
-    }
-  }
-
-  get isOpen() {
-    return !this.closed;
+  const handleClose = () => {
+    setOpen(false);
+    onClose();
   }
 
-  get isClosed() {
-    const { open, opening, closing } = this.state;
-    return !open && !opening && !closing;
-  }
+  const handleContentClick = event => event.stopPropagation();
 
-  get isControlled() {
-    return 'open' in this.props;
-  }
+  const handleBackdropClick = () => {
+    if (!props.closeOnBackdropClick) return;
 
-  open() {
-    this.setState(
-      {
-        opening: true,
-      },
-      () => {
-        setTimeout(() => {
-          document.addEventListener('keydown', this.handleEscapeKey);
-
-          this.setState({
-            open: true,
-            opening: false,
-          });
-        }, 250);
-      }
-    );
-  }
-
-  close() {
-    this.setState(
-      {
-        closing: true,
-      },
-      () => {
-        setTimeout(() => {
-          document.removeEventListener('keydown', this.handleEscapeKey);
-
-          this.setState({
-            open: false,
-            closing: false,
-          });
-        }, 250);
-      }
-    );
-  }
-
-  closeInternal() {
-    if (this.isControlled) {
-      this.props.onClose();
-    } else {
-      this.close();
-    }
-  }
+    handleClose();
+  };
 
-  handleEscapeKey = event => {
-    if (!this.props.closeOnEscape) {
-      return;
-    }
+  const handleKeyDown = useCallback(event => {
+    if (!isOpen || !props.closeOnEscape) return;
 
     if (event.keyCode === 27) {
-      this.closeInternal();
+      handleClose();
     }
-  };
+  });
 
-  handleBackdropClick = () => {
-    if (!this.props.closeOnBackdropClick) {
-      return;
+  useEffect(() => {
+    if (open !== isOpen) {
+      setOpen(open);
     }
+  }, [open]);
 
-    this.closeInternal();
-  };
-
-  handleContentClick = event => {
-    event.stopPropagation();
-  };
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyDown);
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  })
```

Comment:
worth conditionally firing with `[isOpen, props.closeOnEscape]`?

Reply:
Just gave that a try. Presented weird behavior that closes both modals and reopens the second when pressing escape